packages/assets-controllers/src/TokenDetectionController.test.ts

```diff
@@ -438,168 +445,122 @@ describe('TokenDetectionController', () => {
-        isDetectionEnabledFromPreferences: true,
-        selectedAddress: '0x1',
-        chainId: ChainId.mainnet,
-      },
-    );
-
-    await tokenDetection.start();
+        messenger: buildTokenDetectionControllerMessenger(controllerMessenger),
+      });
+      await tokenDetection.start();
 
-    expect(getBalancesInSingleCallMock.called).toBe(true);
-    getBalancesInSingleCallMock.reset();
+      expect(getBalancesInSingleCallMock.called).toBe(true);
+      getBalancesInSingleCallMock.reset();
 
-    tokenDetection.stop();
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    await networkStateChangeListener!({
-      providerConfig: { chainId: toHex(polygonDecimalChainId) },
+      tokenDetection.stop();
+      changeNetwork({
+        chainId: toHex(polygonDecimalChainId),
+        type: NetworkType.rpc,
+        ticker: NetworksTicker.rpc,
+      });
+      expect(getBalancesInSingleCallMock.called).toBe(false);
     });
 
-    expect(getBalancesInSingleCallMock.called).toBe(false);
-  });
-
-  it('should not call getBalancesInSingleCall if onTokenListStateChange is called with an empty token list', async () => {
-    const stub = sinon.stub();
-    const getBalancesInSingleCallMock = sinon.stub();
-    // TODO: Replace `any` with type
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let tokenListStateChangeListener: (state: any) => void;
-    const onTokenListStateChange = sinon.stub().callsFake((listener) => {
-      tokenListStateChangeListener = listener;
-    });
-    tokenDetection = new TokenDetectionController(
-      {
-        onTokenListStateChange,
+    it('should not be called if TokenListController is updated to have an empty token list', async () => {
+      tokenDetection = new TokenDetectionController({
+        chainId: ChainId.mainnet,
         onPreferencesStateChange: stub,
-        onNetworkStateChange: stub,
         getBalancesInSingleCall: getBalancesInSingleCallMock,
         addDetectedTokens: stub,
-        getTokensState: stub,
-        getTokenListState: stub,
-        getNetworkState: () => defaultNetworkState,
+        getTokensState: () => tokensController.state,
         getPreferencesState: () => preferences.state,
-        getNetworkClientById: jest.fn(),
-      },
-      {
-        disabled: false,
-        isDetectionEnabledForNetwork: true,
-        isDetectionEnabledFromPreferences: true,
-      },
-    );
+        messenger: buildTokenDetectionControllerMessenger(controllerMessenger),
+      });
 
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    await tokenListStateChangeListener!({ tokenList: {} });
+      tokenList.clearingTokenListData();
+      expect(getBalancesInSingleCallMock.called).toBe(false);
+    });
 
-    expect(getBalancesInSingleCallMock.called).toBe(false);
-  });
+    it('should be called if onPreferencesStateChange is called with useTokenDetection being true and is changed', async () => {
```

Comment:
Perhaps to be more accurate this should say:

``` suggestion
    it('should be called if onPreferencesStateChange is called with useTokenDetection being true and selectedAddress is changed', async () => {

Reply:
Fixed here: https://github.com/MetaMask/core/pull/3609/commits/f5e66114a14b07cf443eb75e338f99428907b1fb